dask/dataframe/io.py

```diff
@@ -724,10 +738,47 @@ def one_path_one_key(path, key, start, stop, columns, chunksize, lock):
     if (start != 0 or stop is not None) and len(keys) > 1:
         raise NotImplementedError(read_hdf_error_msg)
     from .multi import concat
-    return concat([one_path_one_key(path, k, start, s, columns, chunksize, lock)
+    return concat([one_path_one_key(path, k, start, s, columns, chunksize,
+                                    lock, division_column)
                    for k, s in zip(keys, stops)])
 
 
+def _divide_buffer(buffer, division_column, start, stop, chunksize):
+    divisions = list()
+    starts = [start]
+    stops=list()
+    if isinstance(division_column, string_types):
+        _div_col = lambda x: x[division_column]
+    elif division_column:
+        _div_col = lambda x: pd.Series(x.index)
+    for i, chunk in enumerate(buffer):
```

Comment:
Does it read through the full dataset when we ask to create the dask.dataframe though?